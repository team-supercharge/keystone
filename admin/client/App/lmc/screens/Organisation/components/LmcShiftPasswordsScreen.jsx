import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { ActionCreators } from '../../../actions/actions'
import { connect } from 'react-redux'
import LmcSpinner from '../../../components/LmcSpinner.jsx'
import LmcShiftPasswordsTable from './LmcShiftPasswordsTable.jsx'
import LmcCreateButton from '../../../components/LmcCreateButton.jsx'

export class LmcShiftPasswordsScreen extends Component {
    render() {
        const { shifts, fetchShifts } = this.props

        return (
            <div style={styles.mainContainer}>
                <div style={styles.content}>
                    <LmcCreateButton
                        buttonText='Shift Password'
                        listId='Shift'
                        title='Add a Shift Password'
                        onCreate={fetchShifts}
                        style={styles.addButton}
                    />
                    { shifts ? (
                        <LmcShiftPasswordsTable
                            shifts={shifts}
                        />
                    ) : (
                        <LmcSpinner />
                    )}
                </div>
            </div>
        )
    }
}

const styles = {
    addButton: {
        width: 200,
        float: 'right',
    },
    mainContainer: {
        padding: '50px 20px 0px 20px',
        overflow: 'scroll',
        height: '83vh',
    },
    content: {
        maxWidth: 1000,
        margin: '0 auto',
        wordWrap: 'break-word',
    }
}

LmcShiftPasswordsScreen.propTypes = {
    shifts: PropTypes.array,
    fetchShifts: PropTypes.func.isRequired,
}

const mapStateToProps = state => {
    return {
        shifts: state.data.shifts
    }
}

const mapDispatchToProps = dispatch => {
    return {
        fetchShifts: () => dispatch(ActionCreators.loadList('shifts'))
    }
}

export default connect(mapStateToProps, mapDispatchToProps)(LmcShiftPasswordsScreen)