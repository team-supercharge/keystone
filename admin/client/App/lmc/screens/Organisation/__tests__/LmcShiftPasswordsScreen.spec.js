jest.mock('../../../../shared/CreateForm')

import React from 'react'
import { shallow } from 'enzyme'
import { LmcShiftPasswordsScreen } from '../components/LmcShiftPasswordsScreen.jsx'

describe('LmcShiftPasswordsScreen', () => {
    let wrapper
    let shifts
    const fetchShiftsMock = jest.fn()
    const deleteShiftMock = jest.fn()

    beforeEach(() => {
        shifts = [
            { id: 'testId1', title: 'testShift1', password: 'testPassword1' },
            { id: 'testId2', title: 'testShift2', password: 'testPassword2' }
        ]

        wrapper = shallow(
            <LmcShiftPasswordsScreen 
                shifts={shifts}
                fetchShifts={fetchShiftsMock}
                deleteShift={deleteShiftMock}
            />
        )
    })

    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    it('renders an LmcShiftPasswordsList', () => {
        expect(wrapper.find('LmcShiftPasswordsList').length).toEqual(1)
    })

    it('renders an LmcCreateButton', () => {
        const button = wrapper.find('LmcCreateButton')
        const { buttonText, listId, title, onCreate } = button.props()

        expect(buttonText).toEqual('Shift')
        expect(listId).toEqual('Shift')
        expect(title).toEqual('Add a Shift')
        expect(onCreate).toEqual(fetchShiftsMock)
    })

    it('renders an LmcSpinner when there is no data', () => {
        const emptyWrapper = shallow(
            <LmcShiftPasswordsScreen
                shifts={null}
                fetchShifts={fetchShiftsMock}
                deleteShift={deleteShiftMock}
            />
        )

        expect(wrapper.find('LmcSpinner').length).toEqual(0)
        expect(emptyWrapper.find('LmcShiftPasswordsTable').length).toEqual(0)
        expect(emptyWrapper.find('LmcSpinner').length).toEqual(1)
    })
})