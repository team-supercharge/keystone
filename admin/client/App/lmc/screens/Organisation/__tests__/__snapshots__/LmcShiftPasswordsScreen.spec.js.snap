// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LmcShiftPasswordsScreen renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    style={
      Object {
        "height": "83vh",
        "overflow": "scroll",
        "padding": "50px 20px 0px 20px",
      }
    }
  >
    <div
      style={
        Object {
          "margin": "0 auto",
          "maxWidth": 1000,
          "wordWrap": "break-word",
        }
      }
    >
      <LmcCreateButton
        buttonText="Shift Password"
        listId="Shift"
        onCreate={[MockFunction]}
        style={
          Object {
            "float": "right",
            "width": 200,
          }
        }
        title="Add a Shift Password"
      />
      <LmcShiftPasswordsTable
        shifts={
          Array [
            Object {
              "id": "testId1",
              "password": "testPassword1",
              "title": "testShift1",
            },
            Object {
              "id": "testId2",
              "password": "testPassword2",
              "title": "testShift2",
            },
          ]
        }
      />
    </div>
  </div>,
  "nodes": Array [
    <div
      style={
        Object {
          "height": "83vh",
          "overflow": "scroll",
          "padding": "50px 20px 0px 20px",
        }
      }
    >
      <div
        style={
          Object {
            "margin": "0 auto",
            "maxWidth": 1000,
            "wordWrap": "break-word",
          }
        }
      >
        <LmcCreateButton
          buttonText="Shift Password"
          listId="Shift"
          onCreate={[MockFunction]}
          style={
            Object {
              "float": "right",
              "width": 200,
            }
          }
          title="Add a Shift Password"
        />
        <LmcShiftPasswordsTable
          shifts={
            Array [
              Object {
                "id": "testId1",
                "password": "testPassword1",
                "title": "testShift1",
              },
              Object {
                "id": "testId2",
                "password": "testPassword2",
                "title": "testShift2",
              },
            ]
          }
        />
      </div>
    </div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <LmcShiftPasswordsScreen
        fetchShifts={[MockFunction]}
        shifts={
          Array [
            Object {
              "id": "testId1",
              "password": "testPassword1",
              "title": "testShift1",
            },
            Object {
              "id": "testId2",
              "password": "testPassword2",
              "title": "testShift2",
            },
          ]
        }
      />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": LmcShiftPasswordsScreen {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {
          "fetchShifts": [MockFunction],
          "shifts": Array [
            Object {
              "id": "testId1",
              "password": "testPassword1",
              "title": "testShift1",
            },
            Object {
              "id": "testId2",
              "password": "testPassword2",
              "title": "testShift2",
            },
          ],
        },
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          style={
            Object {
              "height": "83vh",
              "overflow": "scroll",
              "padding": "50px 20px 0px 20px",
            }
          }
        >
          <div
            style={
              Object {
                "margin": "0 auto",
                "maxWidth": 1000,
                "wordWrap": "break-word",
              }
            }
          >
            <LmcCreateButton
              buttonText="Shift Password"
              listId="Shift"
              onCreate={[MockFunction]}
              style={
                Object {
                  "float": "right",
                  "width": 200,
                }
              }
              title="Add a Shift Password"
            />
            <LmcShiftPasswordsTable
              shifts={
                Array [
                  Object {
                    "id": "testId1",
                    "password": "testPassword1",
                    "title": "testShift1",
                  },
                  Object {
                    "id": "testId2",
                    "password": "testPassword2",
                    "title": "testShift2",
                  },
                ]
              }
            />
          </div>
        </div>,
        "_debugID": 2,
        "_renderedOutput": <div
          style={
            Object {
              "height": "83vh",
              "overflow": "scroll",
              "padding": "50px 20px 0px 20px",
            }
          }
        >
          <div
            style={
              Object {
                "margin": "0 auto",
                "maxWidth": 1000,
                "wordWrap": "break-word",
              }
            }
          >
            <LmcCreateButton
              buttonText="Shift Password"
              listId="Shift"
              onCreate={[MockFunction]}
              style={
                Object {
                  "float": "right",
                  "width": 200,
                }
              }
              title="Add a Shift Password"
            />
            <LmcShiftPasswordsTable
              shifts={
                Array [
                  Object {
                    "id": "testId1",
                    "password": "testPassword1",
                    "title": "testShift1",
                  },
                  Object {
                    "id": "testId2",
                    "password": "testPassword2",
                    "title": "testShift2",
                  },
                ]
              }
            />
          </div>
        </div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <LmcShiftPasswordsScreen
    fetchShifts={[MockFunction]}
    shifts={
      Array [
        Object {
          "id": "testId1",
          "password": "testPassword1",
          "title": "testShift1",
        },
        Object {
          "id": "testId2",
          "password": "testPassword2",
          "title": "testShift2",
        },
      ]
    }
  />,
}
`;
